src/error.rs

```diff
@@ -64,6 +64,11 @@ pub enum Error {
         found: i32,
     },
 
+    /// Returned when a comparison operation/value map fails to parse into a `Comparison` struct
+    ComparisonParsingFailed {
```

Comment:
Per my comments where this error is used, it may be possible to use existing error variants instead of creating a new one. However, if you think that overloads the existing errors with meanings that aren't closely enough related, I don't object to creating a new one.